bench.c

```diff
@@ -83,12 +83,16 @@ static void *bench_worker(__attribute__((unused)))
 
     /* wait until all workers created */
     pthread_mutex_lock(&worker_lock);
-    while (!ready)
-        if (pthread_cond_wait(&worker_wait, &worker_lock)) {
-            puts("pthread_cond_wait failed");
-            exit(-1);
-        }
+    if (++ready == MAX_THREAD)
+        pthread_cond_broadcast(&worker_wait);
+    else
+        while (ready < MAX_THREAD)
+            if (pthread_cond_wait(&worker_wait, &worker_lock)) {
+                puts("pthread_cond_wait failed");
+                exit(-1);
+            }
     pthread_mutex_unlock(&worker_lock);
+    /* all workers are ready, let's start bombing kecho */
```

Comment:
Don't mention `kecho` since the benchmark can be used for arbitrary TCP servers.